scripts/vet.sh

```diff
@@ -184,4 +184,7 @@ revive -formatter plain ./... >"${REV_OUT}" || true
 # TODO: Remove `|| true` to unskip linter failures once existing issues are fixed.
 (noret_grep -v "unused-parameter" "${REV_OUT}" | not grep -v "\.pb\.go:") || true
 
+# Error if trailing spaces found in any files excluding files in .git directory and *.md files
+$(find . -path ./.git -prune -o -type f ! -name "*.md" -exec grep -Hn "[[:blank:]]$" {} \;) | fail_on_output
```

Comment:
Why surround the command in `$()`? This will cause the output of grep to be executed, which isn't what we want. I believe we can remove the surrounding `$()`
```
find . -path ./.git -prune -o -type f ! -name "*.md" -exec grep -Hn "[[:blank:]]$" {} \; | fail_on_output
```

However, this will not show all the violations because `fail_on_output` is implemented to exit early with a non-zero exit code if it sees any output. So it closes the read side of the pipe causing errors similar to the following if grep is still writing more results.
```sh
find . -path ./.git -prune -o -type f ! -name "*.md" -exec grep -Hn "[[:blank:]]$" {} \; | fail_on_output   
./features/advancedtls/creds/server_cert.pem:52:            X509v3 Subject Key Identifier: 
./features/advancedtls/creds/server_cert.pem:54:            X509v3 Authority Key Identifier: 
./features/advancedtls/creds/server_cert.pem:56:            X509v3 Basic Constraints: 
./features/advancedtls/creds/server_cert.pem:58:            X509v3 Key Usage: 
./features/advancedtls/creds/server_cert.pem:60:            X509v3 Subject Alternative Name: 
find: ‘grep’ terminated by signal 13
find: ‘grep’ terminated by signal 13
```
A workaround would be to define a version of `fail_on_output` that reads the entire input before exiting, the following code from an LLM demonstrates this:
```sh
fail_on_output() {                                                                                       
  output=$(tee /dev/stderr) # Read the entire output
  if [ -n "$output" ]; then
    return 1
  fi
}
```

Reply:
Yeah missed it. Good catch. The current command does show the violations though